server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -154,6 +155,66 @@ func fillInvocationFromCacheStats(cacheStats *capb.CacheStats, ti *tables.Invoca
 	ti.TotalCachedActionExecUsec = cacheStats.GetTotalCachedActionExecUsec()
 }
 
+func invocationStatusLabel(ti *tables.Invocation) string {
+	if ti.InvocationStatus == int64(inpb.Invocation_COMPLETE_INVOCATION_STATUS) {
+		if ti.Success {
+			return "success"
+		}
+		return "failure"
+	}
+	if ti.InvocationStatus == int64(inpb.Invocation_DISCONNECTED_INVOCATION_STATUS) {
+		return "disconnected"
+	}
+	return "unknown"
+}
+
+func recordCacheMetrics(ti *tables.Invocation) {
```

Comment:
recordCacheMetrics is called from finalizeInvocation, so the status will always be COMPLETE. Unless there's a strong reason to keep it, I would remove InvocationStatusLabel from all of the CacheEvents.

I think this is also an argument for moving the cache event tracking code from here into hit-tracker -- that way you can increment these counters when the action happens.